docs/paper/dev/api/command-api/arguments/predicate.mdx

```diff
@@ -0,0 +1,100 @@
+                final CommandSender sender = ctx.getSource().getSender();
+
+                for (int i = 0; i < 5; i++) {
+                    sender.sendRichMessage("Is <index> in bounds? <result>",
+                        Placeholder.unparsed("index", Integer.toString(i)),
+                        Placeholder.unparsed("result", Boolean.toString(doubleRangeProvider.range().test((double) i)))
+                    );
+                }
+                return Command.SINGLE_SUCCESS;
+            }))
+        .build();
+}
+```
+
+### In-game preview
+<FullWidthVideo src={DoubleRangeMp4}/>
+
+
+## Integer range argument
+This argument works very similarly to the double range argument, with the only difference being that this argument only accepts integers.
+
+### Example usage
+```java title="MinecraftArguments.java"
+public static LiteralCommandNode<CommandSourceStack> integerRangeArgument() {
+    return Commands.literal("integerrange")
+        .then(Commands.argument("range", ArgumentTypes.integerRange())
+            .then(Commands.argument("tested_integer", IntegerArgumentType.integer())
+                .executes(MinecraftArguments::runIntegerRangeCommand)))
+        .build();
+}
+
+private static int runIntegerRangeCommand(final CommandContext<CommandSourceStack> ctx) {
+    final IntegerRangeProvider integerRangeProvider = ctx.getArgument("range", IntegerRangeProvider.class);
+    final int integerToTest = IntegerArgumentType.getInteger(ctx, "tested_integer");
+
+    if (integerRangeProvider.range().contains(integerToTest)) {
+        ctx.getSource().getSender().sendRichMessage("<aqua><input></aqua> <green>is</green> inside the specified range!",
+            Placeholder.unparsed("input", Integer.toString(integerToTest))
+        );
+        return Command.SINGLE_SUCCESS;
+    }
+
+    ctx.getSource().getSender().sendRichMessage("<aqua><input></aqua> <red>is not</red> inside the specified range!",
+        Placeholder.unparsed("input", Integer.toString(integerToTest))
+    );
+
+    return Command.SINGLE_SUCCESS;
+}
+```
+
+### In-game preview
+<FullWidthVideo src={IntegerRangeMp4}/>
+
+
+## Item predicate argument
+This argument allows for checking whether an item fits some predicate. It is useful for filtering out certain items based on some criteria.
+
+### Example usage
+```java
+public static LiteralCommandNode<CommandSourceStack> itemPredicateArgument() {
+    return Commands.literal("itempredicate")
+        .then(Commands.argument("predicate", ArgumentTypes.itemPredicate())
+            .executes(ctx -> {
+                final ItemStackPredicate predicate = ctx.getArgument("predicate", ItemStackPredicate.class);
+                final ItemStack defaultWoodenSword = ItemType.WOODEN_SWORD.createItemStack();
+
+                ctx.getSource().getSender().sendRichMessage("Is default wooden sword? <result>.",
+                    Placeholder.parsed("result", predicate.test(defaultWoodenSword) ? "<green>true" : "<red>false")
```

Comment:
Would be better if the tags were closed to not include the dot or just delete the dot but even then closed tags is better in this context